### PR TITLE
🎨  Change the UI of the Nodes Panel

### PR DESCRIPTION
**Pull request by BHesseldieck**
A refactor of the nodes panel. Goals are:

- Group nodes into categories, so we can make use of the newly created node codex metadata. Additionally, group Core nodes into Subcategories.
 
- Guide users on what to do when they find no results (i.e. no native node for their app)

### CODE REVIEW

File: packages/workflow/src/Interfaces.ts [hunk 1]
```diff
@@ -783,3 +784,9 @@ export interface IRawErrorObject {
 export interface IStatusCodeMessages {
 	[key: string]: string;
 }
+
+export type CodexCategories = {
```
Comment (ivov) [inline packages/workflow/src/Interfaces.ts:788]: Maybe we should rename to `CodexProperties` now that `alias` is part of this.

File: packages/workflow/src/Interfaces.ts [hunk 2]
```diff
@@ -567,6 +567,7 @@ export interface INodeTypeDescription {
 		deactivate?: INodeHookDescription[];
 	};
 	webhooks?: IWebhookDescription[];
+	codex?: CodexCategories;
```
Comment (ivov) [inline packages/workflow/src/Interfaces.ts:570]: @mutdmour If no contents, should `codex` point to an empty object or not be there at all? `codex` here set as optional, but the ticket says `codex` is always required.

File: packages/cli/src/LoadNodesAndCredentials.ts [hunk 3]
```diff
@@ -202,14 +202,58 @@ class LoadNodesAndCredentialsClass {
 		};
 	}
 
+	/**
+	 * Retrieves `categories` and `subcategories` (if defined)
+	 * from the codex for the node at the given file path.
```
Comment (ivov) [inline packages/cli/src/LoadNodesAndCredentials.ts:207]: Should be updated to include `alias`.

File: packages/workflow/src/Interfaces.ts (hunk 2 above)
Comment (mutdmour) [reply packages/workflow/src/Interfaces.ts]: optional makes sense. 👍🏽

**Comment by janober** [comment]
Amazing work. Got merged.

**Comment by janober** [comment]
Got released with n8n@0.125.0

**Pull request merged**